pkg/catalog/diagram.go

```diff
@@ -223,21 +259,28 @@ func (p *Generator) CreateReturnDataModel(appname string, stmnt *sysl.Statement,
 // It handles recursively getting the related types, or for primitives, just returns the
 func (p *Generator) CreateTypeDiagram(appName string, typeName string, t *sysl.Type, recursive bool) string {
 	m := p.RootModule
-	var plantumlString string
-	if recursive {
-		relatedTypes := catalogdiagrams.RecursivelyGetTypes(appName, map[string]*sysl.Type{typeName: NewTypeRef(appName, typeName)}, m)
-		plantumlString = catalogdiagrams.GenerateDataModel(appName, relatedTypes)
-		if _, ok := p.RootModule.GetApps()[appName]; !ok {
-			return ""
-		}
-		// Handle Empty
-		if len(relatedTypes) == 0 {
-			return ""
+	if !p.Mermaid {
+		var plantumlString string
+		if recursive {
```

Comment:
This logic should be in a different function; eg split ` CreateTypeDiagram(` to delegate just like what was done with `CreateSequenceDiagram`

Reply:
Alright

Reply:
Addressed